### PR TITLE
Feature/nodejs basics

### PR DESCRIPTION
**Pull request by bhavinbabariya**

### CODE REVIEW

File: NodeJS-Basics/file-operations/auth.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const express = require("express");
+const fs = require("fs");
+var bcrypt = require("bcryptjs");
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/file-operations/auth.js]: Any specific reason to use **var** over here? If not change it to const

File: NodeJS-Basics/internationalization/PostData.js
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/internationalization/PostData.js:1]: Remove Empty Files

File: NodeJS-Basics/internationalization/auth.js
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/internationalization/auth.js:1]: Change **var** to **const** if var is not needed

File: NodeJS-Basics/practicle2/nextOpen.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    let i;
+    let shop_status;
+    for (i = 1; i < 8; i++) {
+        shop_status = data.find((obj) => obj.day === weekdays[index + i]);
+        if (shop_status) {
+            break;
+        }
+    }
+
+    // console.log("shop status : ", shop_status);
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/practicle2/nextOpen.js]: remove unwanted comments

File: NodeJS-Basics/session-authentication/PostData.js
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/session-authentication/PostData.js:1]: remove empty files

File: NodeJS-Basics/session-authentication/auth.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const express = require("express");
+const fs = require("fs");
+var bcrypt = require("bcryptjs");
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/session-authentication/auth.js]: change var to const

File: NodeJS-Basics/session-authentication/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
+const express = require("express");
+const helmet = require("helmet");
+const cors = require("cors");
+
+const fs = require("fs");
+const path = require("path");
+const app = express();
+var session = require("express-session");
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/session-authentication/index.js]: change var to cont or let

File: NodeJS-Basics/session-authentication/validateUser.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const jwt = require("jsonwebtoken");
+const secret = "authsecret";
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/session-authentication/validateUser.js]: get secret from env, we can not put secrets or passwords in codebase

**Review by jay-m-simformsolutions — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,15 @@
 Calculator/.vscode
 CRUD/.vscode
-SPA/.vscode
\ No newline at end of file
+SPA/.vscode
```
Comment (jay-m-simformsolutions) [inline .gitignore]: use wildcard ** for node_modules as **/node_modules and see if that works, instead of declaring all separately.

**Review by jay-m-simformsolutions — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (bhavinbabariya) [reply .gitignore]: **/.vscode
**/node_modules
**/.env

I have written like above but  still .env files are uploaded to remote repository

File: NodeJS-Basics/practicle1/index.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    "EAST (Eastern African Time)",
+    "WET (Western European Time)",
+    "CAT (Central Africa Time)",
+    "EAT (Eastern Africa Time)",
+    "AST (Arabia Standard Time)",
+    "BTT (Bhutan Time)",
+    "NPT (Nepal Time)",
+];
+
+inquirer
```
Comment (jay-m-simformsolutions) [inline NodeJS-Basics/practicle1/index.js:29]: use async await for this one

**Review by jay-m-simformsolutions — APPROVED** [review]

**Pull request merged**